### PR TITLE
standalone benchmarks to compare random forest, decision trees, and logistic regression

### PR DESCRIPTION
**Pull request by xadupre**

### CODE REVIEW

**Comment by wenbingl** [comment]
@jeffsaremi , can we use this benchmark for spark-ml as well?

File: benchmarks/bench_plot_onnxruntime_decision_tree.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
+# coding: utf-8
+"""
+Benchmark of onnxruntime on DecisionTree.
+"""
+# Authors: Xavier Dupré (benchmark)
+# License: MIT
```
Comment (prabhat00155) [inline benchmarks/bench_plot_onnxruntime_decision_tree.py]: Have we agreed on this style? Shall I adopt this in my changes as well?

File: benchmarks/bench_plot_onnxruntime_decision_tree.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                    # measures the baseline
+                    st = time()
+                    r = 0
+                    for X in Xs:
+                        p1 = fct1(X)
+                        r += 1
+                        if time() - st >= 1:
+                            break  # stops if longer than a second
+                    end = time()
+                    obs["time_skl"] = (end - st) / r
```
Comment (prabhat00155) [inline benchmarks/bench_plot_onnxruntime_decision_tree.py]: Can r be 0 (if Xs is [])?

File: benchmarks/bench_plot_onnxruntime_logreg.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+                    # creates different inputs to avoid caching in any ways
+                    Xs = []
+                    for r in range(repeat):
+                        x = np.empty((n, nfeat))
+                        x[:, :] = rand(n, nfeat)[:, :]
+                        Xs.append(x)
+
+                    # measures the baseline
+                    st = time()
+                    r = 0
```
Comment (prabhat00155) [inline benchmarks/bench_plot_onnxruntime_logreg.py]: Also, what is r? Could the names be a little more descriptive?

File: benchmarks/bench_plot_onnxruntime_logreg.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+        {"name": "onnx", "version": onnx.__version__},
+        {"name": "onnxruntime", "version": onnxruntime.__version__},
+        {"name": "skl2onnx", "version": skl2onnx.__version__},
+    ])
+    df.to_csv("bench_plot_onnxruntime_logreg.time.csv", index=False)
+    print(df)
+    df = run_bench(verbose=True)
+    plt.savefig("bench_plot_onnxruntime_logreg.png")
+    df.to_csv("bench_plot_onnxruntime_logreg.csv", index=False)
+    # plt.show()
```
Comment (prabhat00155) [inline benchmarks/bench_plot_onnxruntime_logreg.py]: Why is this commented code left here?

File: benchmarks/bench_plot_onnxruntime_random_forest.py [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+        {"name": "onnx", "version": onnx.__version__},
+        {"name": "onnxruntime", "version": onnxruntime.__version__},
+        {"name": "skl2onnx", "version": skl2onnx.__version__},
+    ])
+    df.to_csv("bench_plot_onnxruntime_decision_tree.time.csv", index=False)
+    print(df)
+    df = run_bench(verbose=True)
+    plt.savefig("bench_plot_onnxruntime_random_forest.png")
+    df.to_csv("bench_plot_onnxruntime_random_forest.csv", index=False)
+    # plt.show()
```
Comment (prabhat00155) [inline benchmarks/bench_plot_onnxruntime_random_forest.py]: Commented code left behind.

File: benchmarks/bench_plot_onnxruntime_decision_tree.py (hunk 1 above)
Comment (xadupre) [reply benchmarks/bench_plot_onnxruntime_decision_tree.py]: It is a proposal. scikit-learn does that.

File: benchmarks/bench_plot_onnxruntime_decision_tree.py (hunk 2 above)
Comment (xadupre) [reply benchmarks/bench_plot_onnxruntime_decision_tree.py]: X cannot be [] unles the user specifies a wrong parameter in the benchmark. I did not want to add too many extra tests.

File: benchmarks/bench_plot_onnxruntime_logreg.py (hunk 4 above)
Comment (xadupre) [reply benchmarks/bench_plot_onnxruntime_logreg.py]: up to the user, I don't think that makes a big difference.

File: benchmarks/bench_plot_onnxruntime_random_forest.py (hunk 5 above)
Comment (xadupre) [reply benchmarks/bench_plot_onnxruntime_random_forest.py]: done

File: benchmarks/bench_plot_onnxruntime_logreg.py (hunk 4 above)
Comment (xadupre) [reply benchmarks/bench_plot_onnxruntime_logreg.py]: done

File: benchmarks/bench_plot_onnxruntime_logreg.py (hunk 3 above)
Comment (xadupre) [reply benchmarks/bench_plot_onnxruntime_logreg.py]: renamed into repeated

**Pull request merged**